File: src/main/java/uss/code/course/infra/SearchKeywordSanitizer.java
package uss.code.course.infra;

import lombok.experimental.UtilityClass;

import java.util.regex.Pattern;

@UtilityClass
public class SearchKeywordSanitizer {

    private static final Pattern BOOLEAN_OPERATORS = Pattern.compile("[+\\-<>()~*\"@]");
    private static final Pattern WHITESPACES = Pattern.compile("\\s+");
    private static final String SPACE = " ";

    public static String sanitize(final String keyword) {
        final String withoutOperators = BOOLEAN_OPERATORS.matcher(keyword).replaceAll(SPACE);

        return WHITESPACES.matcher(withoutOperators).replaceAll(SPACE).trim();
    }
}

File: src/main/kotlin/uss/code/course/infra/SearchKeywordSanitizer.kt
package uss.code.course.infra

object SearchKeywordSanitizer {
    private val BOOLEAN_OPERATORS = Regex("""[+\-<>()~*"@]""")
    private val WHITESPACES = Regex("""\s+""")
    private const val SPACE = " "

    fun sanitize(keyword: String): String {
        val withoutOperators = BOOLEAN_OPERATORS.replace(keyword, SPACE)

        return WHITESPACES.replace(withoutOperators, SPACE).trim { it <= ' ' }
    }
}
